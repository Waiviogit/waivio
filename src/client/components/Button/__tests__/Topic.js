import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import Topic from '../Topic';

describe('<Topic />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      name: 'exampletopic',
      favorite: true,
      closable: true,
    };

    wrapper = mount(
      <Router>
        <Topic {...props} />
      </Router>,
    );

    act(() => {
      wrapper.update();
    });
  });

  it('renders without exploding', () => {
    wrapper = shallow(<Topic {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should renred i tag', () => {
    expect(wrapper.find('i')).not.toBeUndefined();
  });

  it('renders include Topic--closing in className', () => {
    wrapper.find('i').simulate('mouseover');
    expect(wrapper.find('Link').props().className).toBe('Topic Topic--favorite Topic--closing');
  });

  it('renders include Topic--favorite in className', () => {
    expect(wrapper.find('Link').props().className).toBe('Topic Topic--favorite');
  });
});
