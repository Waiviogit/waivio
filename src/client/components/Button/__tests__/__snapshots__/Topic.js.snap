// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topic /> renders without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Topic
    closable={true}
    favorite={true}
    name="exampletopic"
    onClose={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        "#",
        "exampletopic",
        <i
          className="iconfont icon-close Topic__close"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          role="presentation"
        />,
      ],
      "className": "Topic Topic--favorite",
      "to": "/trending/exampletopic",
    },
    "ref": null,
    "rendered": Array [
      "#",
      "exampletopic",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "iconfont icon-close Topic__close",
          "onClick": [Function],
          "onMouseOut": [Function],
          "onMouseOver": [Function],
          "role": "presentation",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "#",
          "exampletopic",
          <i
            className="iconfont icon-close Topic__close"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            role="presentation"
          />,
        ],
        "className": "Topic Topic--favorite",
        "to": "/trending/exampletopic",
      },
      "ref": null,
      "rendered": Array [
        "#",
        "exampletopic",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "iconfont icon-close Topic__close",
            "onClick": [Function],
            "onMouseOut": [Function],
            "onMouseOver": [Function],
            "role": "presentation",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
